src/main/java/uk/gov/digital/ho/hocs/casework/security/UserPermissionsService.java

```diff
@@ -32,31 +33,54 @@ public UUID getUserId() {
     public AccessLevel getMaxAccessLevel(String caseType) {
         Set<PermissionDto> permissionDtos = getUserPermission();
         Optional<PermissionDto> maxPermission = permissionDtos.stream()
-                .filter(e-> e.getCaseTypeCode().equals(caseType))
+                .filter(e -> e.getCaseTypeCode().equals(caseType))
                 .max(Comparator.comparing(PermissionDto::getAccessLevel));
         return maxPermission.orElse(
-               new PermissionDto("", AccessLevel.UNSET)
+                new PermissionDto("", AccessLevel.UNSET)
         ).getAccessLevel();
     }
 
-
     public Set<UUID> getUserTeams() {
         String[] groups = requestData.groupsArray();
-        return Arrays.stream(groups)
+
+        return Stream.of(groups)
                 .map(this::getUUIDFromBase64)
                 .filter(Objects::nonNull)
                 .collect(Collectors.toSet());
     }
 
-    public Set<String> getUserCaseTypes() {
-        return getUserPermission().stream()
-                .map(PermissionDto::getCaseTypeCode)
-                .collect(Collectors.toSet());
+    public boolean isUserInTeam(UUID teamUuid) {
+        return getExpandedUserTeams().contains(teamUuid);
+    }
+
+    public Set<UUID> getExpandedUserTeams() {
+        var userTeamUuids = getUserTeams();
 
+        if (userTeamUuids.isEmpty()) {
+            return userTeamUuids;
+        }
 
+        var allTeams = infoClient.getTeams();
+
+        var userCaseAdminCaseTypes = getUserCaseAdminCaseTypes(allTeams, getUserTeams());
```

Comment:
not blocking, but can we call getUserTeams once in this method?

Reply:
I put it above for reuse, not sure why I didn’t change this. Will sort it later!

Reply:
Changed...